mizar/dp/mizar/operators/endpoints/endpoints_operator.py

```diff
@@ -239,16 +258,13 @@ def produce_simple_endpoint_interface(self, ep):
             interfaces = InterfaceServiceClient(
                 ep.get_droplet_ip()).ProduceInterfaces(InterfacesList(interfaces=interfaces_list))
 
-        # At this point Mizar has provisioned the network
-        # TODO (cathy): mark the pod network as ready! Shall it be here or in
-        # the pod builtin wf.
         logger.info("Produced {}".format(interfaces))
 
     def create_simple_endpoints(self, interfaces, spec):
         """
         Create a simple endpoint object (calling the API operator)
         """
-        for interface in interfaces.interfaces:
+        for interface, net_info in zip(interfaces.interfaces, spec['interfaces']):
```

Comment:
Instead of looking in spec['interfaces'] again here, can you just add  the needed information to the interface message/object?

See here.
https://github.com/futurewei-cloud/mizar/blob/dev-next/mizar/dp/mizar/operators/endpoints/endpoints_operator.py#L313

Reply:
This not going to work tho, https://github.com/clu2xlu/mizar/blob/cni-delete-new/mizar/dp/mizar/workflows/builtins/pods/create.py#L60
This is the interfaces defined in the yaml file when calling kubectl create.